### PR TITLE
GHA: setup GitHub actions for windows

### PR DESCRIPTION
**Pull request by erw7**
#12960 continued.

For now, the build will succeed. ~This has never passed a test with `MSVC_32` (flaky test?).~

### CODE REVIEW

File: .github/workflows/ci_windows.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      - uses: actions/checkout@v2
+
+      - uses: actions/cache@v2
+        with:
+          path: C:\projects\nvim-deps
+          key: ${{ matrix.config }}-${{ hashFiles('third-party\**') }}
+
+      - uses: actions/setup-python@v2
+        with:
+          pyhon-version: '2.7'
```
Comment (jamessan) [inline .github/workflows/ci_windows.yml:28]:
```suggestion
          python-version: '2.7'
```

File: .github/workflows/ci_windows.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+      - uses: actions/setup-python@v2
+        with:
+          pyhon-version: '3.5'
```
Comment (jamessan) [inline .github/workflows/ci_windows.yml:32]:
```suggestion
          python-version: '3.5'
```

File: .github/workflows/ci_windows.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    name: Build and test
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/cache@v2
+        with:
+          path: C:\projects\nvim-deps
+          key: ${{ matrix.config }}-${{ hashFiles('third-party\**') }}
+
+      - uses: actions/setup-python@v2
```
Comment (jamessan) [inline .github/workflows/ci_windows.yml:26]: I'm not sure we actually need to use these actions.  Their intent, as I understand it, is to ensure the specified Python version is provided in `%PATH%`.  Is the second one actually additive or does it override the first?

`build.ps1` was also updated to use the full paths to the existing tools, so I think that's also an argument for these being unnecessary.

File: .github/workflows/ci_windows.yml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+      - uses: actions/setup-node@v1
+        with:
+          node-version: '12.x'
+
+      - name: Run CI
+        run: powershell ci\build.ps1
+        env:
+          CONFIGURATION: ${{ matrix.config }}
+
+      - name: Upload artifacts
```
Comment (jamessan) [inline .github/workflows/ci_windows.yml:44]: The CI workflow doesn't need to upload artifacts.  That's part of the release workflow.

File: .github/workflows/ci_windows.yml [hunk 5]
```diff
@@ -0,0 +1,52 @@
+name: Run Windows CI
```
Comment (jamessan) [inline .github/workflows/ci_windows.yml:1]: Could this be merged into `.github/workflows/ci.yml` as a separate job?  Something like

```yaml
jobs:
  unixish:
    ... existing Linux/macOS setup ...
  windows:
    ... Windows CI setup ...
```

File: ci/build.ps1 [hunk 6]
```diff
@@ -113,16 +87,16 @@ elseif ($compiler -eq 'MSVC') {
 
 if (-not $NoTests) {
   # Setup python (use AppVeyor system python)
-  C:\Python27\python.exe -m pip install pynvim ; exitIfFailed
-  C:\Python35\python.exe -m pip install pynvim ; exitIfFailed
+
+  C:\hostedtoolcache\windows\Python\2.7.18\x64\python.exe -m pip install pynvim ; exitIfFailed
```
Comment (jamessan) [inline ci/build.ps1:91]: Might be better to use [Get-ToolcacheToolDirectory](https://github.com/actions/virtual-environments/blob/fc8402e007bc3b7ae4a449edd57dee7a151a331e/images/win/scripts/ImageHelpers/InstallHelpers.ps1#L297-L301) or `$env:AGENT_TOOLSDIRECTORY`.

**Comment by Sebastian-byte** [comment]
I Aprove This.

File: .github/workflows/ci_windows.yml (hunk 3 above)
Comment (erw7) [reply .github/workflows/ci_windows.yml]: I tried it and these were not necessary. It seems to be just a matter of adding the path to the beginning of  `%PATH%`.

File: ci/build.ps1 (hunk 6 above)
Comment (erw7) [reply ci/build.ps1:91]: I tried and found that they are not defined in the virtual environment. Isn't this for use when creating virtual images? If anyone is an expert in GHA and knows a good way to get these paths from within a virtual environment, please let me know.

File: .github/workflows/ci_windows.yml (hunk 5 above)
Comment (erw7) [reply .github/workflows/ci_windows.yml]: I tried it. For some reason, msgpack is no longer found in the MINGW_32 build. Hmm, that's about it for today...

**Comment by erw7** [comment]
I haven't tried it, but it may be possible to install `Neovim::Ext` using `perl` installed under `C:\hostedtoolcache\windows`. However, the perl provider is not tested on Windows, as shown below. So, I will remove the steps to install `Neovim::Ext`.

https://github.com/neovim/neovim/blob/5ce328df401bc5cafd66caeb265835b939028b7f/test/functional/provider/perl_spec.lua#L26-L27

Also, in my environment, `:perl print('Hello')` was not returned on Windows, and an infinite loop seemed to occur. Therefore, I don't think perl provider will work on WIndows at present. The current CI was supposed to install `Neovim::Ext`, but was that because the perl provider is now running on Windows?

**Comment by janlazo** [comment]
@erw7 `:perl` won't work on Windows but some code from the perl provider works. The initial  perl provider test, based on a nodejs provider test, worked at one point on Windows but the test is not stable enough to run on every build. Skip the perl provider for now in Github Actions.

**Pull request merged**